From 902a43a2a1ad574bb195c759fc853ae5752adc27 Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Wed, 29 Jan 2020 00:51:30 +0200
Subject: [PATCH] [SW-8228] goya2: add init of qspi and spi write limit

Adding to init the required value for the qspi_baudr register which
sets the devisor of the controller and set write limit to 16MB on spi
flash

Change-Id: I60c29e0622adc888451914d78147c099f743cb49
---

diff --git a/app/arcboot_rom.c b/app/arcboot_rom.c
index ec82dbf..0dfdf54 100644
--- a/app/arcboot_rom.c
+++ b/app/arcboot_rom.c
@@ -14,7 +14,11 @@
 	.config_fuse = 0,
 };
 
-static void arcboot_rom_init(void) {}
+static void arcboot_rom_init(void)
+{
+	regmap_writel(SPI_FLASH_BAUDR_DIVISOR, mmPSOC_QSPI_BAUDR);
+	regmap_writel(write_en_16MB, mmPSOC_SECURITY_SPI_WRITE_LIMIT);
+}
 
 static void arcboot_rom_fuse(void)
 {
diff --git a/common/spi_flash.h b/common/spi_flash.h
index 9e99bef..894f58c 100644
--- a/common/spi_flash.h
+++ b/common/spi_flash.h
@@ -8,9 +8,23 @@
 #ifndef _SPI_FLASH_H_
 #define _SPI_FLASH_H_
 
+enum flash_write_limit {
+	write_en_64K,
+	write_en_128KB,
+	write_en_256KB,
+	write_en_512KB,
+	write_en_1MB,
+	write_en_2MB,
+	write_en_4MB,
+	write_en_8MB,
+	write_en_16MB
+};
+
 #define ENABLE_BYTE_SWAP		1
 #define DISABLE_BYTE_SWAP		0
 
+#define SPI_FLASH_BAUDR_DIVISOR		2
+
 #define FLASH_PRIMARY_OFFSET		0
 #define FLASH_SECONDARY_OFFSET		4
 #define FLASH_EFUSE_OFFSET		8
